### PR TITLE
feat(theme): add support for meta og locale and alternates

### PR DESCRIPTION
**Pull request by FlorinaPacurar**
## Pre-flight checklist

- [X] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [X] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Fixes issue #8887 by adding og locale and alternates

## Test Plan

Tested locally in browser with suggested config
Screenshot 2023-07-17 at 17 13 04

### Test links

Deploy preview: https://deploy-preview-9152--docusaurus-2.netlify.app/

## Related issues/PRs

issue #8887

### CODE REVIEW

**Review by slorber — CHANGES_REQUESTED** [review]
Thanks, almost looks good.

This should rather be handled in `theme/SiteMetadata` (theme) instead of `SiteMetadataDefaults` because supporting og locale attributes is opinionated, shouldn't be mandatory for users and we should give them the ability to remove support if they don't want it (which would reduce the size of the html pages).

See location of original PR here: https://github.com/facebook/docusaurus/pull/8894

**Comment by FlorinaPacurar** [comment]
Thanks @slorber - updated.

File: packages/docusaurus-theme-classic/src/theme/SiteMetadata/index.tsx [hunk 1]
```diff
@@ -24,10 +24,18 @@ import SearchMetadata from '@theme/SearchMetadata';
 // See https://github.com/facebook/docusaurus/issues/3317
 function AlternateLangHeaders(): JSX.Element {
   const {
-    i18n: {defaultLocale, localeConfigs},
+    i18n: {currentLocale, defaultLocale, localeConfigs},
   } = useDocusaurusContext();
   const alternatePageUtils = useAlternatePageUtils();
 
+  const currentHtmlLang = localeConfigs[currentLocale]?.htmlLang;
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/theme/SiteMetadata/index.tsx:31]:
```suggestion
  const currentHtmlLang = localeConfigs[currentLocale]!.htmlLang;
```

We can assume each locale will have a config in the map

File: packages/docusaurus-theme-classic/src/theme/SiteMetadata/index.tsx [hunk 2]
```diff
@@ -24,10 +24,18 @@ import SearchMetadata from '@theme/SearchMetadata';
...
+
+  // HTML lang is a BCP 47 tag, but the Open Graph protocol requires
+  // using underscores instead of dashes.
+  // See https://ogp.me/#optional
+  // See https://en.wikipedia.org/wiki/IETF_language_tag)
+  const bcp47ToLocale = (code: string): string => code.replace('-', '_');
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/theme/SiteMetadata/index.tsx:37]:
```suggestion
  const bcp47ToOpenGraphLocale = (code: string): string => code.replace('-', '_');
```

More explicit fn name

File: packages/docusaurus-theme-classic/src/theme/SiteMetadata/index.tsx [hunk 3]
```diff
@@ -51,6 +59,19 @@ function AlternateLangHeaders(): JSX.Element {
         })}
         hrefLang="x-default"
       />
+
+      {currentHtmlLang && (
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/theme/SiteMetadata/index.tsx:63]: Now `currentHtmlLang` can't be undefined so this condition is useless

**Review by slorber — CHANGES_REQUESTED** [review]
Looks good 👍 just requesting minor code changes

**Review by slorber — APPROVED** [review]
LGTM thanks 👍

**Pull request merged**